### ISSUE TITLE
Using boost 1.69 with /CLR (windows .net) fails

### ISSUE THREAD

**Issue by jbulow** [boostorg/system#33]
Triggers this error:

```
Warning	C4561	
'__fastcall' incompatible with the '/clr' option: converting to '__stdcall'
...\include\boost\type_traits\detail\is_member_function_pointer_cxx_11.hpp	120	

Error	C2953	
'boost::is_member_function_pointer<Ret(__stdcall C::* )(Args...)>': class template has already been defined
...\include\boost\type_traits\detail\is_member_function_pointer_cxx_11.hpp	120	

```

The fix is attached.
[clr_patch.txt](https://github.com/boostorg/system/files/2861930/clr_patch.txt)

**Comment by pdimov** [comment]
The inclusion of what Boost headers triggers this error for you?

**Comment by jbulow** [comment]
Yes. I just upgraded from boost 1.67 and got this problem with 1.69.

**Comment by jbulow** [comment]
Any header that includes the affected files I guess. I don't know an easy way to find out how an include files was included in the first place when using visual studio. I don't think it does matter. It is an obvious error with an obvious fix. The _MANAGED guard could be narrowed down to the _fastcall declarations maybe.

**Comment by pdimov** [comment]
Yes but the error is in Type Traits and your fix refers to Type Traits, but you've opened an issue against Boost.System, which is why I was wondering whether there's something to fix in Boost.System.

I'll transfer the issue to Type Traits then.
